Document `required_recipe` feature and configuration

danielsdeleo opened the pull request:
As described in the change, this feature is intended for expert use
only, therefore it should be okay to just document it with the optional
settings.

References:

* https://discourse.chef.io/t/chef-12-20-3-released/10857
* https://github.com/chef/chef-rfc/blob/master/rfc089-server-enforced-recipe.md

The feature also requires Chef Server 12.15+:
https://discourse.chef.io/t/chef-server-12-15-0-released/10851

Signed-off-by: Daniel DeLeo <dan@chef.io>

jamescott commented:
Hi Mike,

This will be in chef-client 12.7 release?

mikedodge04 commented:
@jamescott I'm still waiting to hear about (chef/chef#4111), maybe you can help me?

jamescott commented:
I'm gonna wait to update this until the chef#4111 calms down a bit more. I have some questions about the Ruby types for the Apple-specific parameters (they don't seem to match up -- but that may well be as intended). Once that PR is set for approval, I'll finish up the doc for this. No worries.

jamescott commented:
https://docs.chef.io/release/12-8/resource_launchd.html

^^ current draft for 12-8

natewalck commented:
@jamescott , It looks like some wires are crossed.

resource_osx_profile is completely different from resource_launchd.

It looks like some terminology and actions from osx_profile: https://docs.chef.io/release/12-7/resource_osx_profile.html

Ended up in the documentation for resource_launchd:

https://docs.chef.io/release/12-8/resource_launchd.html

A launchd is *not* a configuration profile. It is a completely different resource. The way it is referred to in this PR is correct (  "...resource to manage launchd daemons / agents") but in the documentation you linked, it says "A launchd resource to manage configuration profiles (.mobileconfig files) on the Mac OS X platform", which is **not** correct.

For instance:

resource_osx_profile is a .mobileconfig file

resource_launchd is a .plist file

A resource_launchd is a service like cron or upstart. It is not a configuration profile at all. It is a service.

Thanks for getting the docs updated! Let us know if any further clarification is needed.

jamescott commented:
Here's the doc for launchd: https://docs.chef.io/release/12-8/resource_launchd.html

Please put any new issues or PRs on the `chef-web-docs` repo (note that this one has a `-2016` appended to it).

Thanks for the docs notes and the new resource!

james

natewalck commented:
@jamescott I've made an issue on chef-web-docs (https://github.com/chef/chef-web-docs/issues/18). This **really** needs to be fixed. The docs do not match what Mike has in this PR, they were changed for some reason.

The docs submitted here were correct, so I am unsure why they got changed in the actual commit.

danielsdeleo committed:
Document `required_recipe` feature and configuration

As described in the change, this feature is intended for expert use
only, therefore it should be okay to just document it with the optional
settings.

References:

* https://discourse.chef.io/t/chef-12-20-3-released/10857
* https://github.com/chef/chef-rfc/blob/master/rfc089-server-enforced-recipe.md

The feature also requires Chef Server 12.15+:
https://discourse.chef.io/t/chef-server-12-15-0-released/10851

Signed-off-by: Daniel DeLeo <dan@chef.io>

danielsdeleo commented:
https://github.com/chef/chef-server/pull/1214 also has some relevant info/discussion

Commit:
adds server version note

brewn commented:
![tenor-237149116](https://user-images.githubusercontent.com/21040805/28144464-9b8ef8da-6739-11e7-97a1-06e7f63e67a6.gif)
